base/abstractdict.jl

```diff
@@ -335,6 +335,75 @@ function _typeddict(d::AbstractDict, others::AbstractDict...)
     Dict{K,V}(d)
 end
 
+function reduce(::typeof(merge), dicts::AbstractVector{T}) where T <: AbstractDict{K,V} where {K,V}
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return _merge!(empty(first(dicts)),dicts) # use `empty`. `similar` makes routine 20x slower
+end
+
+function reduce(::typeof(merge), d::T, dicts::AbstractVector{T}) where T <: AbstractDict{K,V} where {K,V}
+    isempty(dicts) && return copy(d)
+    return _merge!(empty(first(dicts)),d,dicts) # There is no evidence that calling _merge! twice is less efficient.
+end
+
+# This catches typeof(T) == UnionAll.
+# The conversion of `dicts` to a vector of type Any makes this method much faster.
+function reduce(::typeof(merge), dicts::AbstractVector{T}) where T <: AbstractDict
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return reduce(merge, Any[x for x in dicts])
+end
+
+function reduce(::typeof(merge), d::AbstractDict, dicts::AbstractVector{T}) where T <: AbstractDict
+    isempty(dicts) && throw(ArgumentError("Cannot merge empty array"))
+    return reduce(merge, d, Any[x for x in dicts])
+end
+
+function reduce(m::typeof(merge), items::AbstractVector)
+    isempty(items) && throw(ArgumentError("Cannot merge empty vector"))
+    length(items) == 1 && return copy(first(items))
+    i1 = firstindex(items) + 1
+    i2 = lastindex(items)
+    return reduce(m,first(items), view(items,i1:i2))
```

Comment:
`i2` can be replaced with `end`.

Reply:
Good. I wasn't sure about it.

Reply:
Are you sure about replacing `i2` with `end`?
```
julia> a = collect(1:10);
julia> view(a, 1:end);
ERROR: syntax: missing last argument in "1:" range expression 
julia> view(a, 1:lastindex(a));

julia> ex = Expr(:call,:getindex, :a, Expr(:call,:(:),1,:end))
:(getindex(a, 1:end))
julia> eval(ex)
ERROR: UndefVarError: end not defined
```
Now I recall having tried `end` first. Must be a good reason for this.

Reply:
Thanks. That will work. Anyway the method in question has been removed.
I just rebased and pushed all the changes in your last round of comments.